### ISSUE TITLE
FEATURE - [Add Purchase Orders to Xero packages]

### ISSUE THREAD

**Issue by DJFresh66** [fivetran/dbt_xero#21]
**Are you a Fivetran customer?** 

Yes, Doug Jeffery, Principal Data Architect, Modulous

**Is your feature request related to a problem? Please describe.**

The Modulous FD has asked me if we can report on Purchase Orders in Xero - I can see they are synched from Xero into Snowflake but they are not considered in the dbt transformations as far as I can tell

**Describe the solution you'd like**
 
Extend the fivetran dbt xero and dbt xero source packages to create a transformed view of Purchase Orders

**Describe alternatives you've considered**
 
Extending the packages myself

**Additional context**

**Please indicate the level of urgency and business impact of this request** 

It would keep my FD happy but not super urgent tbh

**Are you interested in contributing to this package?** 

- [x] Yes, I can do this and open a PR for your review.
- [ ] Possibly, but I'm not quite sure how to do this. I'd be happy to do a live coding session with someone to get this work implemented. 
- [ ] No, I'd prefer if someone else did this.  I don't have the time and/or don't know how to incorporate the changes necessary.

**Comment by fivetran-joemarkiewicz** [comment]
Thanks so much for opening this feature request @DJFresh66!!

I agree that I think adding purchase order reporting to the Xero package would be a great addition! What type of reporting around Purchase Orders would you be looking to add?

Realistically, our team will be able to scope out work for this feature come January. However, I see you are open to contributing to the package! We (along with the community of people using this package) would be incredibly grateful if you wanted to contribute. Otherwise, we can chat more here about the vision of the model and our team can scope it out come January 😄

**Comment by DJFresh66** [comment]
Hi Joe, I would like to contribute to the package if possible.

Full disclosure, I’m pretty new to dbt (though very experienced in data / SQL / BI etc overall) so I am essentially looking to copy what you have already done and adapt it. I have put together a mini spec doc attached to see if I am on the right lines 😊

What would be the next step / process for me to contribute?

Cheers,

Doug

***@***.***
Doug Jeffery  Principal Data Architect
+44 (0) 2035 821 005 | +44 (0) 7900 496 912
modulous.com<http://modulous.com/>


Modulous Limited - Company No. 11523631 - Registered Office at Greenworks, Dog & Duck Yard, Princeton Street, London, WC1R 4BH, UK.  This email (and any attachments) is confidential for the intended recipient(s) only. If you are not the recipient(s) please contact us and delete this email immediately.

**Comment by fivetran-joemarkiewicz** [comment]
That's great to hear @DJFresh66!

I think the next best step would be to schedule some time with myself and the team using our [office hours](https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours?month=2021-12) link. From there, we can go over your spec and how best for you to integrate it into the dbt package. I'll keep an eye out for the calendar invite 👀 

Looking forward to chatting more with you!

**Comment by DJFresh66** [comment]
Hi @fivetran-joemarkiewicz 

FYI I have completed developing and testing my changes to the xero source and xero packages. Before I raise PRs I need to discuss an issue and get your input on how to handle it:

- The latest code on github for both packages includes changes to support dbt 1.0.0 so these came along when I forked the repos
- I am not set up to work locally with dbt, only on dbt cloud, I could not find a way to emulate 'local' packages on dbt cloud so I ended up specifying the package dependencies using - git pointing to my forked repos
- This nearly worked but for some reason I just couldn't get the package dependencies sorted out using dbt cloud, dbt clean kept crashing and dpt deps kept coming up with different errors around dbt_utils etc.
- Long story short, I ended up making changes in both my repos to packages.yml and dbt_project.yml to make them look like they used to in V3 using dbt 0.21.0 and previous version of fivetran_utils then testing in the fivetran environment 

So, I can raise PRs with the code as is and you can work out how to merge the changes and bring the dbt version and package dependencies back in line or I can revert the yml back in line with the latest but either way it won't have been tested 'properly' with dbt v1.0.0.

Let me know what you would like me to do,

Cheers, Doug

**Comment by fivetran-joemarkiewicz** [comment]
Hi @DJFresh66!

I have heard from other users when making changes like this within dbt cloud have experienced some weirdness since `dbt clean` is not a supported command in dbt cloud. We really appreciate your work in developing this enhancement to the dbt_xero package. My team and I handle these dbt compatibility issues frequently and we would be happy to take the next steps in your PR for you once you open it 😄 
> Please note however, the next release of the dbt_xero package will have to be compatible with dbt v1.0.0. 

Once you open the PR, I will be sure to review it and let you know if I have any questions. Thanks again!

**Comment by DJFresh66** [comment]
Cool, thanks @fivetran-joemarkiewicz I will be raising the PRs (one for xero, one for xero_source) asap 👍
